libs/external-router/routing/src/guard/existence.guard.ts

```diff
@@ -54,12 +63,40 @@ export class DaffExternalRouterExistenceGuard implements CanActivate {
     state: RouterStateSnapshot,
   ): Observable<UrlTree | boolean> {
     return this.driver.resolve(this.urlNormalizer.normalize(state.url)).pipe(
-      switchMap(resolvedRoute => of(this.externalRouter.add(resolvedRoute))),
+      map((resolvedRoute: DaffExternallyResolvableUrl) => processErrors(resolvedRoute)),
+      switchMap(resolvedRoute => {
+        this.externalRouter.add(resolvedRoute);
+        return of(resolvedRoute);
+      }),
+      map((resolvedRoute: DaffExternallyResolvableUrl) => processRedirects(resolvedRoute)),
       // Note that we have to use state.url as we want to ensure that we keep any fragments or query strings.
       // When we succeed, redirect to the new route.
       map(() => this.router.parseUrl(state.url)),
-      // When we fail, navigate to the configured failed resolution path,
+      // When we fail, we need to process errors as dictated by the error type.
+      // Handle redirects
       catchError(error => {
+        if(
+          !(error instanceof DaffExternalRouterPermanentRedirectError) &&
+          !(error instanceof DaffExternalRouterTemporaryRedirectError)) {
+          throw error;
+        }
+
+        //On platform browser, we need to add a redirect for the original url and redirect to the new url.
+        this.externalRouter.addRedirect(state.url, error.redirectUrl);
```

Comment:
`state.url` may not be safe to insert here as it contains queries and hashes.

Reply:
yeah because that means that the path will only match with the query strings, right?

Reply:
Fixed in abe8b76cc3adb9b8ac0ef4e98dff25987494542e -- @griest024 can you give it a once over to verify.